nvtabular/loader/tensorflow.py

```diff
@@ -246,19 +249,7 @@ def __getitem__(self, idx):
         with Keras model.fit. Does not leverage
         passed idx in any way
         """
-        try:
-            return DataLoader.__next__(self)
-        except StopIteration:
-            # TODO: I would like to do a check for idx == 0
-            # here, but that requires that tf.keras.Model.fit
-            # be called with shuffle=False, and that seems
-            # small enough that it would be too easy to miss
-            # for many users. That said, blind reinitialization
-            # is probably irresponsible, so worth thinking
-            # of something better here
-            # return StopIteration
-            DataLoader.__iter__(self)
```

Comment:
It makes sense that we don't want to create a new iterator and return more data here, but do we still want to catch `StopIteration`?

Reply:
yeah I agree that is why I moved it out of there...?

Reply:
Ah, the possibility of a `StopIteration` still exists but we're going to bubble it out to the calling code? :+1: